components/patina_smbios/src/component.rs

```diff
@@ -0,0 +1,384 @@
+    }
+}
+
+impl SmbiosProvider {
+    /// Create a new SMBIOS provider
+    pub fn new() -> Self {
+        Self {}
+    }
+
+    /// Initialize the SMBIOS provider and register it as a service
+    #[coverage(off)] // Component integration - tested via integration tests
+    fn entry_point(
+        self,
+        config: Config<SmbiosConfiguration>,
+        mut commands: Commands,
+        boot_services: StandardBootServices,
+    ) -> Result<()> {
+        let cfg = (*config).clone();
+
+        // Create the manager with configured version
+        let manager = SmbiosManager::new(cfg.major_version, cfg.minor_version).map_err(|e| {
+            log::error!("Failed to create SMBIOS manager: {:?}", e);
+            patina::error::EfiError::Unsupported
+        })?;
+
+        // Convert boot_services to 'static by leaking it
+        // SAFETY: This is safe because boot_services remains valid for the entire lifetime
+        // of the system. We leak it intentionally to get a 'static reference.
+        let boot_services_static: &'static StandardBootServices = Box::leak(Box::new(boot_services));
+
+        // Wrap the manager in TplMutex for TPL-aware reentrancy protection at TPL_NOTIFY level
+        let manager_mutex = Box::leak(Box::new(TplMutex::new(boot_services_static, Tpl::NOTIFY, manager)));
+
+        // Install the C/EDKII protocol - this installs a global reference for C compatibility
+        if let Err(e) = install_smbios_protocol(manager_mutex, boot_services_static) {
+            log::error!("Failed to install SMBIOS protocol: {:?}", e);
+        }
+
+        // Register unified SMBIOS service with all operations
+        commands.add_service(Smbios { manager: manager_mutex });
+
+        Ok(())
+    }
+}
+
+/// SMBIOS service providing all SMBIOS operations
+///
+/// This service provides a complete interface for SMBIOS table management:
+/// - **Type-safe operations**: `add_record<T>()` for structured record types
+/// - **Byte-level operations**: `add_from_bytes()`, `update_string()`, `remove()`
+/// - **Table management**: `version()`, `publish_table()`
+///
+/// All operations are protected by a TplMutex at TPL_NOTIFY level for thread safety.
+///
+/// # Example
+///
+/// ```ignore
+/// fn entry_point(
+///     smbios: Service<Smbios>,
+///     boot_services: StandardBootServices,
+/// ) -> Result<()> {
+///     // Add structured records
+///     smbios.add_record(None, &bios_info)?;
+///     
+///     // Publish to configuration table
+///     smbios.publish_table(&boot_services)?;
+///     Ok(())
+/// }
+/// ```
+#[derive(IntoService, Clone, Copy)]
+#[service(Smbios)]
+pub struct Smbios {
+    manager: &'static TplMutex<'static, SmbiosManager, StandardBootServices>,
+}
+
+impl Smbios {
+    /// Gets the SMBIOS version information.
+    ///
+    /// # Returns
+    ///
+    /// A tuple of (major_version, minor_version).
+    pub fn version(&self) -> (u8, u8) {
+        let manager = self.manager.lock();
```

Comment:
version should be outside of the lock at a minimum. or using something other then the TplMutex. I just merged a PR that saw an up-to 200% performance improvement by doing somethign similar, due to the overhead of the Tpl Mutex.

https://github.com/OpenDevicePartnership/patina/pull/947